src/File.ts

```diff
@@ -125,10 +132,26 @@ const tryRepetitive = async ({
         }
 
         const line = fileLines[listItemCache.position.start.line];
-
         if (line.includes(globalFilter)) {
             if (sectionIndex === originalTask.sectionIndex) {
-                listItem = listItemCache;
+                const dateFromFileName = new Lazy(() => DateFallback.fromPath(originalTask.path));
+                const taskFromLine = Task.fromLine({
+                    line,
+                    path: originalTask.path,
+                    precedingHeader: originalTask.precedingHeader,
+                    sectionStart: originalTask.sectionStart,
+                    sectionIndex: originalTask.sectionIndex,
```

Comment:
using these fields from originalTask makes sense. 

we locate the line with path, sectionStart, and sectionIndex, therefore those are part of the new Task that gets created from the line. The only cause for concern here is precedingHeader as that may change in the interim.

is there something you're saying that I'm not understanding?

Reply:
> is there something you're saying that I'm not understanding?

Maybe, or maybe vice versa. Maybe I'm not understanding. And unfortunately the earlier conversations are all hidden now, by being resolved.

I thought that the idea behind calling `task.identicalTo()` was to check more fields in the two tasks, but it looks to me like one of the tasks is populated from fields from the other task.

So it looks to me like `task.identicalTo()` is giving a false sense of extra value - and it would probably be safer and clearer to check `task.originalMarkdown`.

I may later refactor the new code out to a helper function in a file that does not import Obsidian types, to be able to write some tests for it, and explore the behaviour better.

> The only cause for concern here is precedingHeader as that may change in the interim.

Yes, that is a concern.

Another concern is a change behind the scenes somehow from this:

```text
## header

- [ ] task 1
- [ ] task 2 - I am the task that will be completed
```

to this:

```text
## header

- [ ] task 1
- [ ] some new task that was added - so the old 2nd task is now the 3rd task in the section
- [ ] task 2 - I am the task that will be completed
```

Reply:
have you read my comment here: https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1663#issuecomment-1430710631? 

it contains the answer to the purpose of `task.identitcalTo`

I could elaborate more if needed.

Reply:
Yes I did see that. I should have said that I don’t understand why the rendering of the task matters.

The line being overwritten should be identical to the original text read in to the task being edited. 

If the two markdown lines differ in any way, including order of fields, something has gone wrong as the file has been edited in some way behind Tasks’ back, and I do not feel it is safe to overwrite the line.

Reply:
Hi @BluBloos Thank you, and I totally understand... I'll change it to use `.originalMarkdown` when I tweak the error message...